src/main/java/bitflow4j/script/CompilationException.java

```diff
@@ -54,4 +54,19 @@ public static String formatError(Token position, RuleContext ctx, String msg) {
         return errors;
     }
 
+    private static String concatErrors(List<String> errors) {
+        if (errors.isEmpty())
+            return "no errors";
+        else if (errors.size() == 1)
+            return errors.get(0);
```

Comment:
Just for discussion. is get(0) the same like getIterator().next() when list is not empty?

Reply:
Yes, same effect, but in my opinion more explicit than getIterator().next(). Especially when the size is asserted directly above the get(0). However, get(0) only works for lists, not for Sets etc.